### ISSUE TITLE
Lint error with set/project not accepting seqable collection for ks

### ISSUE THREAD

**Issue by frwdrik** [clj-kondo/clj-kondo#2143]
**version**

Using `[com.github.clj-kondo/lein-clj-kondo 2023.07.13]`

**platform**

`Leiningen 2.10.0 on Java 17.0.7 OpenJDK 64-Bit Server VM`

**problem**

The following gives a lint error "Expected: sequential collection, received: set":
```clojure
(set/project [{:foo :bar}] #{:foo})
```

**repro**
```clojure
(ns user
  (:require [clojure.set :as set]))

(set/project [{:foo :bar}] #{:foo})

```
```bash
clj-kondo --lint repro.clj
repro.clj:4:28: error: Expected: sequential collection, received: set.
linting took 22ms, errors: 1, warnings: 0
```
**expected behavior**
I expected no lint errors. The reason being that `set/project` will accept any seqable collection as last argument. The tests for `set/project` in Clojure itself passed a set as last arg: https://github.com/clojure/clojure/blob/501348deb1e49736f3c49f2c177bcff947fe3a17/test/clojure/test_clojure/clojure_set.clj#L142.
Happy to provide a PR :)

**Comment by borkdude** [comment]
Fixed, thanks!
